examples/build_operation/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	v1 "github.com/openshift/api/build/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	buildv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
+	"path/filepath"
+	"flag"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"fmt"
+)
+
+func main() {
+	var kubeconfig *string
+	if home := homeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	buildV1Client, err := buildv1.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	namespace := "testproject"
+	builds, err := buildV1Client.Builds(namespace).List(metav1.ListOptions{})
+
+	if err != nil {
+		panic(err.Error())
+	}
+	fmt.Printf("There are %d builds in the cluster\n", len(builds.Items))
+
+	build := "cakephp-ex-1"
+	myBuild, err := buildV1Client.Builds(namespace).Get(build, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		fmt.Printf("Build %s in namespace %s not found\n", build, namespace)
+	} else if statusError, isStatus := err.(*errors.StatusError); isStatus {
+		fmt.Printf("Error getting build %s in namespace %s: %v\n",
+			build, namespace, statusError.ErrStatus.Message)
+	} else if err != nil {
+		panic(err.Error())
+	} else {
+		fmt.Printf("Found build %s in namespace %s\n", build, namespace)
+		fmt.Printf("Found build %s in status %+v\n", build, myBuild.Status)
+	}
```

Comment:
In addition to the raw printout of the build, can we show how to extract select fields and print them out nicely?  Maybe build name, start time, duration and status/phgase

Reply:
Done